Flaky test: test_integration

This test has been acting flakily on komodo-unit-testing 

https://ci.equinor.com/komodo/job/unit-tests/job/ert/1201/


```
_______________________________ test_integration _______________________________

tmp_path = PosixPath('/tmp/pytest-of-f_scout_ci/pytest-270/test_integration0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f9410e74cc0>

    @pytest.mark.integration_test
    @pytest.mark.requires_ert_storage
    def test_integration(tmp_path, monkeypatch):
        """Actually start the server, wait for it to be online and do a health check"""
        monkeypatch.chdir(tmp_path)
    
        # Note: Sqlite needs at least 4-5 seconds to spin up even on
        # an unloaded M1-based Mac using local disk. On the CI-server
        # we have less control of available resources, so set timeout-
        # value large to allow time for sqlite to get ready
        with Storage.start_server(timeout=120) as server:
            resp = requests.get(
>               f"{server.fetch_url()}/healthcheck", auth=server.fetch_auth()
            )

/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/ert/test-root/tests/ert_tests/services/test_storage_service.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/prog/res/komodo/2022.01.b5-py36-rhel7/root/lib64/python3.6/site-packages/ert_shared/services/storage_service.py:59: in fetch_url
    for url in self.fetch_conn_info()["urls"]:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ert_shared.services.storage_service.Storage object at 0x7f9410e74518>
timeout = 20

    def fetch_conn_info(self, timeout: float = 20) -> Mapping[str, Any]:
        self.wait_until_ready(timeout)
        if self._conn_info is None:
>           raise ValueError("conn_info is None")
E           ValueError: conn_info is None

/prog/res/komodo/2022.01.b5-py36-rhel7/root/lib64/python3.6/site-packages/ert_shared/services/_base_service.py:356: ValueError
```

--- a/ert_shared/services/_base_service.py
+++ b/ert_shared/services/_base_service.py
@@ -327,12 +327,15 @@ class BaseService:
             pass
         return cls.start_server(*args, **kwargs)
 
-    def wait_until_ready(self, timeout: Optional[float] = None) -> bool:
-        self._conn_info_event.wait(timeout)
-        if self._conn_info_event.is_set():
+    def wait_until_ready(self, timeout: int = None) -> bool:
+        if timeout is None:
+            timeout = self._timeout
+
+        if self._conn_info_event.wait(timeout):
             return not (
                 self._conn_info is None or isinstance(self._conn_info, Exception)
             )
+
         return False  # Timeout reached
 
     def wait(self) -> None:
@@ -353,12 +356,14 @@ class BaseService:
 
         self._conn_info_event.set()
 
-    def fetch_conn_info(self, timeout: float = 20) -> Mapping[str, Any]:
-        self.wait_until_ready(timeout)
-        if self._conn_info is None:
-            raise ValueError("conn_info is None")
-        elif isinstance(self._conn_info, Exception):
+    def fetch_conn_info(self) -> Mapping[str, Any]:
+        is_ready = self.wait_until_ready(self._timeout)
+        if isinstance(self._conn_info, Exception):
             raise self._conn_info
+        elif not is_ready:
+            raise TimeoutError()
+        elif self._conn_info is None:
+            raise ValueError("conn_info is None")
         return self._conn_info
 
     def shutdown(self) -> int:

